packages/protocol/test/common/integration.ts

```diff
@@ -408,61 +406,129 @@ contract('Integration: Exchange', (accounts: string[]) => {
+    // Note that this test relies on having purchased cUSD in the previous test.
+    describe('When selling stable token', () => {
+      before(async () => {
+        originalStable = await stableToken.balanceOf(accounts[0])
+        originalGold = await goldToken.balanceOf(accounts[0])
+        originalReserve = await goldToken.balanceOf(reserve.address)
+        await stableToken.approve(exchange.address, sellAmount)
+        // Cannot sell more than was purchased in the previous test.
+        await exchange.sell(sellAmount.div(20), minBuyAmount, false)
+        finalStable = await stableToken.balanceOf(accounts[0])
+        finalGold = await goldToken.balanceOf(accounts[0])
+        finalReserve = await goldToken.balanceOf(reserve.address)
+      })
+
+      it(`should reduce user's stable`, async () => {
+        assert.isTrue(finalStable.lt(originalStable))
+      })
+
+      it(`should increase user's gold`, async () => {
+        assert.isTrue(finalGold.gt(originalGold))
+      })
+
+      it(`should reduce Reserve's gold`, async () => {
+        assert.isTrue(finalReserve.lt(originalReserve))
+      })
     })
   })
 
-  // Note that this test relies on having purchased cUSD in the previous test.
-  describe('When selling stable token', () => {
-    before(async () => {
-      originalStable = await stableToken.balanceOf(accounts[0])
-      originalGold = await goldToken.balanceOf(accounts[0])
-      originalReserve = await goldToken.balanceOf(reserve.address)
-      await stableToken.approve(exchange.address, sellAmount)
-      // Cannot sell more than was purchased in the previous test.
-      await exchange.exchange(sellAmount.div(20), minBuyAmount, false)
-      finalStable = await stableToken.balanceOf(accounts[0])
-      finalGold = await goldToken.balanceOf(accounts[0])
-      finalReserve = await goldToken.balanceOf(reserve.address)
-    })
+  describe('Buying', () => {
+    const buyAmount = new BigNumber(1)
+    const maxSellAmount = new BigNumber('1000000000000000000')
+
+    describe('When buying stable token', () => {
+      before(async () => {
+        originalStable = await stableToken.balanceOf(accounts[0])
+        originalGold = await goldToken.balanceOf(accounts[0])
+        originalReserve = await goldToken.balanceOf(reserve.address)
+        await goldToken.approve(exchange.address, maxSellAmount)
+        await exchange.buy(buyAmount, maxSellAmount, false)
+        finalStable = await stableToken.balanceOf(accounts[0])
+        finalGold = await goldToken.balanceOf(accounts[0])
+        finalReserve = await goldToken.balanceOf(reserve.address)
+      })
 
-    it(`should reduce user's stable`, async () => {
-      assert.isTrue(finalStable.lt(originalStable))
-    })
+      it(`should increase user's stable`, async () => {
+        assert.isTrue(finalStable.gt(originalStable))
+      })
+
+      it(`should reduce user's gold`, async () => {
+        const block = await web3.eth.getBlock('latest')
```

Comment:
Could this miner-checking code be pulled out to a function to deduplicate? Potentially even out to test-utils.ts, since I think this is used in other tests as well.